thunderstorm/src/main/frontend/_ats/ATS_AppConfigEditor/ATS_AppConfigEditor.tsx

```diff
@@ -0,0 +1,135 @@
+import * as React from 'react';
+import {ComponentSync} from '../../core/ComponentSync';
+import {AppConfigKey_FE, DB_AppConfig, ModuleFE_AppConfig} from '../../_entity';
+import {LL_H_C, LL_V_L} from '../../components/Layouts';
+import {AppToolsScreen, TS_AppTools} from '../../components/TS_AppTools';
+import {__stringify, exists, sortArray} from '@nu-art/ts-common';
+import {_className} from '../../utils/tools';
+import './ATS_AppConfigEditor.scss';
+import {TS_JSONViewer} from '../../components/TS_JSONViewer/TS_JSONViewer';
+import {TS_TextArea} from '../../components/TS_Input';
+import {TS_BusyButton} from '../../components/TS_BusyButton';
+
+type State = {
+	configs: DB_AppConfig[];
+	selectedKey?: string;
+	dataString?: string;
+}
+
+export class ATS_AppConfigEditor
+	extends ComponentSync<{}, State> {
+
+	//######################### Static #########################
+
+	static Screen: AppToolsScreen = {
+		key: 'app-config-editor',
+		renderer: this,
+		name: 'App Config Editor',
+		group: 'Editors',
+		modulesToAwait: [ModuleFE_AppConfig],
+	};
+
+	//######################### Life Cycle #########################
+
+	protected deriveStateFromProps(nextProps: {}, state: State) {
+		const allConfigs = ModuleFE_AppConfig.cache.allMutable();
```

Comment:
Need to pass a filter as prop, to apply to app-configs collection.
Can't simply show all of them without any regard, e.g in our project we use app-configs for permissions and we cannot allow people without permission to see or edit permissions, access to them in this editor.
In application code, we need to be able to pass a filter that checks if it's a permission-key, so we can filter it away.

Reply:
This is a deeper issue than just permission keys.. maybe we should explicitly add an option to make a key non editable when we define it in code..

Reply:
I think that allowing application level to decide what can be editable is enough as first phase.
When do we really need to keep a config uneditable?
Ideally, we'd have an app level provided filter that would check for whatever key types it needs to check, and in this filter it'll also check permission to show a config or not. So if you have permission to edit permissions, you will see permission-keys.
But at least provide a filter for now so that when we add it to the project we can add at least a basic filtering of permissions out.

Reply:
This can work out but feels a bit dangerous. It'll require us to remember to edit this app level filter each time we add a new "uneditable" config.. and I'm sure we'll have more than just permissions... Putting this configuration on the key itself will be a part of its creation process and much much safer IMO

Reply:
fair enough, i'll add a filter prop

Reply:
Added prop "appConfigFilter" to filter out app configs.
checked in my local, works...